MO social security deduction calculations are still incorrect

Following the merge of PR #1780, the MO section C calculations are still incorrect.  Here is a test case where the MO tax unit has an AGI of roughly $100,000 but has a taxable income of zero.

```
- name: Tax unit with taxsimid 77665 in f21.its.csv and f21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 41
        employment_income: 18_010
        taxable_interest_income: 5_505
        taxable_pension_income: 500
        social_security: 9_000
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        is_tax_unit_spouse: true
        age: 41
        employment_income: 57_010
        taxable_interest_income: 5_505
        taxable_pension_income: 500
        social_security: 9_000
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person3:
        age: 11
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person4:
        age: 11
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person5:
        age: 16
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person6:
        age: 16
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2, person3, person4, person5, person6]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2, person3, person4, person5, person6]
        premium_tax_credit: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2, person3, person4, person5, person6]
        state_code: MO
  output:  # expected results from patched TAXSIM35 2023-02-13 version
    mo_income_tax: 3095.77
```
The test results are:
```
================================= FAILURES ===================================
_______________________________ test session _________________________________
/Users/mrh/work/Policy-Engine-US/MO/f21-id77665.yaml:
  Test 'Tax unit with taxsimid 77665 in f21.its.csv and f21.ots.csv':
    mo_income_tax@2021: [0.] differs from 3095.77001953125
                       with an absolute margin [3095.77] > 0.01
========================= short test summary info ============================
FAILED f21-id77665.yaml::
============================== 1 failed in 0.69s ===============================
```

--- a/changelog_entry.yaml
+++ b/changelog_entry.yaml
@@ -1,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
+      - MO social security or social security disability deduction calculation.

--- a/policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction/integration_tests/mo_pension_and_ss_or_ssd.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction/integration_tests/mo_pension_and_ss_or_ssd.yaml
@@ -310,3 +310,64 @@
     mo_pension_and_ss_or_ssd_deduction_section_b: [6_000, 6_000]
     mo_pension_and_ss_or_ssd_deduction_section_c: [0, 0]
     mo_pension_and_ss_or_ssd_deduction: 22_000
+
+- name: Tax unit with taxsimid 77665 in f21.its.csv and f21.ots.csv
+  absolute_error_margin: 0.01
+  period: 2021
+  input:
+    people:
+      person1:
+        is_tax_unit_head: true
+        age: 41
+        employment_income: 18_010
+        taxable_interest_income: 5_505
+        taxable_pension_income: 500
+        social_security: 9_000
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+      person2:
+        is_tax_unit_spouse: true
+        age: 41
+        employment_income: 57_010
+        taxable_interest_income: 5_505
+        taxable_pension_income: 500
+        social_security: 9_000
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+      person3:
+        age: 11
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+      person4:
+        age: 11
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+      person5:
+        age: 16
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+      person6:
+        age: 16
+        ssi: 0  # not in TAXSIM35
+        state_supplement: 0  # not in TAXSIM35
+        wic: 0  # not in TAXSIM35
+    spm_units:
+      spm_unit:
+        members: [person1, person2, person3, person4, person5, person6]
+        snap: 0  # not in TAXSIM35
+        tanf: 0  # not in TAXSIM35
+    tax_units:
+      tax_unit:
+        members: [person1, person2, person3, person4, person5, person6]
+        premium_tax_credit: 0  # not in TAXSIM35
+    households:
+      household:
+        members: [person1, person2, person3, person4, person5, person6]
+        state_code: MO
+  output:  # expected results from patched TAXSIM35 2023-02-13 version
+    mo_income_tax: 3095.77

--- a/policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction/integration_tests/mo_pension_and_ss_or_ssd.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction/integration_tests/mo_pension_and_ss_or_ssd.yaml
@@ -25,8 +25,8 @@
     mo_adjusted_gross_income: [35_000, 35_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 10_000]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 10_000]
-    mo_pension_and_ss_or_ssd_deduction: 30_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 0]
+    mo_pension_and_ss_or_ssd_deduction: 20_000
 
 - name:  70k income, childless couple, evenly split income, one with SS, one with SSD
   period: 2021
@@ -55,8 +55,8 @@
     mo_adjusted_gross_income: [35_000, 25_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [20_000, 20_000]
-    mo_pension_and_ss_or_ssd_deduction: 40_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 10_000]
+    mo_pension_and_ss_or_ssd_deduction: 20_000
 
 - name:  70k income, childless couple, evenly split income, both with SS and SSD
   period: 2021
@@ -85,8 +85,8 @@
     mo_adjusted_gross_income: [31_000, 31_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [12_000, 12_000]
-    mo_pension_and_ss_or_ssd_deduction: 24_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [6_000, 6_000]
+    mo_pension_and_ss_or_ssd_deduction: 12_000
 
 - name:  70k income, childless couple, evenly split income, one with SS, one with private pension
   period: 2021
@@ -115,8 +115,8 @@
     mo_adjusted_gross_income: [35_000, 35_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 10_000]
-    mo_pension_and_ss_or_ssd_deduction: 20_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 0]
+    mo_pension_and_ss_or_ssd_deduction: 10_000
 
 - name:  90k income, childless couple, evenly split income, both with public + private pensions, and SS
   period: 2021
@@ -148,8 +148,8 @@
     mo_adjusted_gross_income: [45_000, 45_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [20_000, 20_000]
-    mo_pension_and_ss_or_ssd_deduction: 40_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 10_000]
+    mo_pension_and_ss_or_ssd_deduction: 20_000
 
 - name:  90k income, childless couple, evenly split income, both with public + private pensions, one with SS and the other with SSD 
   period: 2021
@@ -182,8 +182,8 @@
     mo_adjusted_gross_income: [40_000, 40_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [20_000, 20_000]
-    mo_pension_and_ss_or_ssd_deduction: 40_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [10_000, 10_000]
+    mo_pension_and_ss_or_ssd_deduction: 20_000
 
 - name:  100k income, childless couple, evenly split income, both with public + private pensions, both with SS and SSD 
   period: 2021
@@ -216,8 +216,8 @@
     mo_adjusted_gross_income: [46_000, 45_000]
     mo_pension_and_ss_or_ssd_deduction_section_a: [0, 0]
     mo_pension_and_ss_or_ssd_deduction_section_b: [0, 0]
-    mo_pension_and_ss_or_ssd_deduction_section_c: [21_000, 21_000]
-    mo_pension_and_ss_or_ssd_deduction: 42_000
+    mo_pension_and_ss_or_ssd_deduction_section_c: [11_000, 10_000]
+    mo_pension_and_ss_or_ssd_deduction: 21_000
 
 - name:  70k income, childless couple, evenly split income, both with public pensions
   period: 2021

--- a/policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction_section_c.py
+++ b/policyengine_us/variables/gov/states/mo/tax/income/deductions/mo_pension_and_ss_or_ssd_deduction_section_c.py
@@ -16,23 +16,18 @@ class mo_pension_and_ss_or_ssd_deduction_section_c(Variable):
 
     def formula(person, period, parameters):
         tax_unit = person.tax_unit
-        mo_agi = person("mo_adjusted_gross_income", period)
-        tax_unit_mo_agi = tax_unit.sum(mo_agi)
-
+        ind_mo_agi = person("mo_adjusted_gross_income", period)
+        unit_mo_agi = tax_unit.sum(ind_mo_agi)
         filing_status = tax_unit("filing_status", period)
         p = parameters(period).gov.states.mo.tax.income.deductions
-        ss_or_ssd_agi_allowance = p.mo_ss_or_ssd_deduction_allowance[
-            filing_status
-        ]
-        agi_over_ss_or_ssd_allowance = max_(
-            tax_unit_mo_agi - ss_or_ssd_agi_allowance, 0
-        )  # different from Sections A and B, Line 3, floor at 0.
-        tax_unit_taxable_social_security_benefits = add(
-            tax_unit, period, ["taxable_social_security"]
-        )
-
-        return max_(
-            tax_unit_taxable_social_security_benefits
-            - agi_over_ss_or_ssd_allowance,
+        unit_allowance = p.mo_ss_or_ssd_deduction_allowance[filing_status]
+        unit_agi_over_allowance = max_(0, unit_mo_agi - unit_allowance)
+        ind_taxable_ben = person("taxable_social_security", period)
+        unit_taxable_ben = tax_unit.sum(ind_taxable_ben)
+        unit_deduction = max_(0, unit_taxable_ben - unit_agi_over_allowance)
+        ind_frac = where(
+            ind_taxable_ben > 0,
+            ind_taxable_ben / unit_taxable_ben,
             0,
         )
+        return ind_frac * unit_deduction

